src/debugging/python/PythonDebugHelper.ts

```diff
@@ -0,0 +1,196 @@
+import { DockerDebugConfigurationBase } from "../DockerDebugConfigurationBase";
+import { DockerDebugConfiguration } from "../DockerDebugConfigurationProvider";
+import { PythonScaffoldingOptions } from '../DockerDebugScaffoldingProvider';
+
+export interface PythonPathMapping {
+    localRoot: string;
+    remoteRoot: string;
+}
+
+export interface PythonDebugOptions {
+    host?: string;
+    port?: number;
+    pathMappings?: PythonPathMapping[];
+    justMyCode?: boolean;
+    projectType?: PythonProjectType;
+    django?: boolean;
+    jinja?: boolean;
+}
+
+export interface PythonDockerDebugConfiguration extends DockerDebugConfigurationBase {
+    python?: PythonDebugOptions;
+}
+
+export class PythonDebugHelper implements DebugHelper {
+    public constructor(
+        private readonly cliDockerClient: CliDockerClient,
+        private readonly localOsProvider: LocalOSProvider) {
+    }
+
+    public async provideDebugConfigurations(context: DockerDebugScaffoldContext, options?: PythonScaffoldingOptions): Promise<DockerDebugConfiguration[]> {
+        const configs : DockerDebugConfiguration[] =
+        [{
+            name: "Docker Python Launch",
+            type: "docker",
+            request: "launch",
+            preLaunchTask: "docker-run: debug",
+            python: {
+                pathMappings: [
+                    {
+                        /* eslint-disable-next-line no-template-curly-in-string */
+                        localRoot: "${workspaceFolder}",
+                        remoteRoot: "/app"
+                    }
+                ],
+                projectType: options.projectType
+            }
+        }];
+
+        // If we generated compose files, then we should generate a Python attach configuration.
+        if (context.generateComposeTask) {
+            configs.push(
+                {
+                    name: "Python: Remote Attach",
+                    type: "python",
+                    request: "attach",
+                    host: "localhost",
+                    port: PythonDefaultDebugPort,
+                    pathMappings: [
+                        {
+                            /* eslint-disable-next-line no-template-curly-in-string */
+                            localRoot: "${workspaceFolder}",
+                            remoteRoot: "/app"
+                        }
+                    ]
+                });
+        }
+
+        return configs;
+    }
+
+    public async resolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: PythonDockerDebugConfiguration): Promise<ResolvedDebugConfiguration | undefined> {
+        const containerName = inferContainerName(debugConfiguration, context, context.folder.name);
+
+        // Since Python is a special case, we need to ensure the container is removed before attempting to resolve
+        // the debug configuration.
+        try {
```

Comment:
Do we need to test that `removeContainerAfterDebug` is set?

Reply:
At this point, we're removing the container _before_ debugging not after. Which is the same logic we have here: https://github.com/microsoft/vscode-docker/blob/22abd6cf00ad132c2f9af0af3768eb1b6ec9a681/src/debugging/DockerDebugConfigurationProvider.ts#L109